CMakeLists.txt

```diff
@@ -0,0 +1,277 @@
+cmake_minimum_required(VERSION 3.5)
+
+project(
+  libmbus
+  LANGUAGES CXX C
+  VERSION "0.9.0")
+
+if(CMAKE_BUILD_TYPE STREQUAL "")
+  message(STATUS "CMAKE_BUILD_TYPE empty setting to Debug")
+  set(CMAKE_BUILD_TYPE "Debug")
+endif()
+
+# ##############################################################################
+# default options -> changed with e.g. cd build && cmake ..
+# -DLIBMBUS_BUILD_TESTS=ON
+# ##############################################################################
+
+option(LIBMBUS_BUILD_EXAMPLES "build examples" OFF)
+option(LIBMBUS_BUILD_TESTS "build tests" OFF)
+option(LIBMBUS_ENABLE_COVERAGE "build with coverage support" ON)
```

Comment:
Isn't it weird to enable coverage by default?
I would let the CI scripts enable coverage.